extensions/common/context.h

```diff
@@ -213,19 +216,33 @@ flatbuffers::DetachedBuffer extractEmptyNodeFlatBuffer();
 // address access.
 flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer();
 
+// Extract upstream peer metadata from upstream host metadata.
+// Returns true if the metadata is found in the upstream host metadata.
+bool extractPeerMetadataFromUpstreamHostMetadata(
+    flatbuffers::FlatBufferBuilder& fbb);
+
 // Returns flatbuffer schema for node info.
 std::string_view nodeInfoSchema();
 
 class PeerNodeInfo {
  public:
-  PeerNodeInfo(const std::string_view peer_metadata_id_key,
-               const std::string_view peer_metadata_key);
+  explicit PeerNodeInfo(const std::string_view peer_metadata_id_key,
+                        const std::string_view peer_metadata_key);
+  PeerNodeInfo() = delete;
   const ::Wasm::Common::FlatNode& get() const;
-  const std::string& getId() const { return peer_id_; }
+  const std::string& id() const { return peer_id_; }
+
+  // Found indicates whether both ID and metadata is available.
   bool found() const { return found_; }
 
+  // Maybe waiting indicates that the metadata is not found but may arrive
+  // later.
+  bool maybeWaiting() const {
```

Comment:
super nit: would this be clearer as something like: `notReceived` or `Unknown`, to eliminate potential confusion around passive vs active waiting?

Reply:
notReceived or unknown seems to be also confusing. Not received implies `peer_id_ == kMetadataNotFoundValue` instead of `!=`. `unknown` makes more sense to me.

Reply:
It's closer to the usage in plugins, which check whether it's too early to report.

Reply:
Having `wait` there seems good to me, because we don't report and wait to receive for metadata.. something like `waitToReceive`?